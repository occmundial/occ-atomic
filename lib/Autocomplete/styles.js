import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
