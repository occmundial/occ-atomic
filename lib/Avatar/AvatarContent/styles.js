import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
