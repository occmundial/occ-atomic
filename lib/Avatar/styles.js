import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
