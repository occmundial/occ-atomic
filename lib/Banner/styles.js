import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
