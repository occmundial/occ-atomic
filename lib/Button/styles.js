import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
