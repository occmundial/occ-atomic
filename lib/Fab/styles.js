
import colors from '../tokens/colors.json';
import shadows from '../tokens/shadows.json';

export default {
  fab: {
    background: `${colors.bg.surface.default} !important`,
    outline: `1px solid ${colors.border.default.default}`,
    outlineOffset: '-1px',
    boxShadow: shadows["elevation-elevation-4"],
    transition: '0.3s all, 0s outline, 0s outline-offset',
    '&:hover': {
      outlineColor: colors.border.default.bold,
      background: `${colors.bg.default} !important`,
    },
    '&:active, &:focus': {
      outline: `2px solid ${colors.border.default.bold}`,
      outlineOffset: '-2px',
      background: `${colors.bg.default} !important`,
    },
    '&$disabled': {
      color: colors.text.indigo.secondary,
      cursor: 'not-allowed',
      outlineColor: colors.border.default.subtle
    },
    '&:not(:active):focus-visible': {
      boxShadow: shadows['focus-corp'],
      outlineColor: colors.border.default.default,
    },
  },
  disabled: {
    cursor: 'default',
    pointerEvents: 'none',
  },
};
