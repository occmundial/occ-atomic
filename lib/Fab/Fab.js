import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';
/**
 *
 * A floating action button appears in front of all screen content as a circular shape with an icon in its center.
 */

const Fab = props => {
	const {
    size,
		classes,
		className,
		testId,
		icon,
		onClick,
		href,
		target,
		rel,
		disabled,
		id
	} = props;

  let classNames = classes.fab;
  if (className) classNames += ` ${className}`;
	if (disabled) classNames += ` ${classes.disabled}`;

	return (
		<Button
			size={size}
			icon={icon}
			onClick={onClick}
			href={href}
			target={target}
			rel={rel}
			disabled={disabled}
			round
			testId={testId}
			className={classNames}
			id={id}
			theme="tertiary"
		/>
	);
};

Fab.propTypes = {
	classes: PropTypes.object.isRequired,
	/** Use the name of any of the icons available in the library. */
	icon: PropTypes.string.isRequired,
	/** The fab has three different sizes: sm, md, lg. */
	size: PropTypes.oneOf(['sm', 'md', 'lg']),
	/** It disables the button. */
	disabled: PropTypes.bool,
	/** The function to call when the button is clicked. */
	onClick: PropTypes.func,
	/** Optionally, you can set the href property and use it like a link. */
	href: PropTypes.string,
	/** If you use the href property, you can also define the target. */
	target: PropTypes.string,
	/** If you use the href property, you can also define the rel attribute. */
	rel: PropTypes.string,
	id: PropTypes.string,
	className: PropTypes.string,
	/** The testId property adds the data attribute data-testid to the main element and should be used for testing only. */
	testId: PropTypes.string
};

export default Fab;
