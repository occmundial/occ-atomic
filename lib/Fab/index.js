import injectSheet from 'react-jss';
import Fab from './Fab';
import styles from './styles';

export default injectSheet(styles)(Fab);
