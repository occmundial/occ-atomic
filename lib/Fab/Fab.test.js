import React from 'react';
import { shallow, mount } from 'enzyme';
import Fab from './Fab';
import styles from './styles';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe('FAB', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(
			<Fab icon="testing" theme="primary" classes={classes} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('returns a Button with an icon', () => {
		const wrapper = mount(<Fab classes={classes} icon="testing" />);
		expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button span Jss(Icon)').length).toBe(1);
		expect(wrapper.find('a').length).toBe(0);
	});

	it('returns an anchor', () => {
		const wrapper = mount(
			<Fab classes={classes} href="http://occ.com.mx" icon="testing" />
		);
		expect(wrapper.find('button').length).toBe(0);
		expect(wrapper.find('a').length).toBe(1);
	});

	it('has the right theme', () => {
		const wrapper = mount(<Fab classes={classes} icon="testing" />);
    expect(wrapper.find('button').prop('className').indexOf('Button-primary')).not.toBe(-1);
    wrapper.setProps({ theme: 'secondary' });
    expect(wrapper.find('button').prop('className').indexOf('Button-secondary')).not.toBe(-1);
	});

	it('has the right size', () => {
		const wrapper = mount(<Fab classes={classes} icon="testing" size="md" />);
		expect(wrapper.find('button').prop('className').indexOf('md')).not.toBe(-1);
		expect(wrapper.find('button').prop('className').indexOf('lg')).toBe(-1);
		wrapper.setProps({ size: 'lg' });
		expect(wrapper.find('button').prop('className').indexOf('md')).toBe(-1);
		expect(wrapper.find('button').prop('className').indexOf('lg')).not.toBe(-1);
	});
});
