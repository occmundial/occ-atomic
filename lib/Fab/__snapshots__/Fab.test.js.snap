// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB matches the snapshot 1`] = `
<button
  className="Button-btn-0-1-22 Button-btn-0-1-1 Button-tertiary-0-1-27 Button-tertiary-0-1-6 fab Button-iconOnly-0-1-38 Button-iconOnly-0-1-18 Button-round-0-1-40 Button-round-0-1-20"
>
  <span
    className="Button-cont-0-1-23 Button-cont-0-1-2"
  >
    <span
      className="Icon-oldIcon-0-1-45 Icon-oldIcon-0-1-43 Button-icon-0-1-36 Button-icon-0-1-16"
    />
    
  </span>
</button>
`;
